ai/ai-react-app/src/components/Layout/LeftSidebar.tsx

```diff
@@ -20,7 +25,74 @@ const LeftSidebar: React.FC<LeftSidebarProps> = ({
   setActiveMode,
   activeBackend,
   setActiveBackend,
+  generativeParams,
+  setGenerativeParams,
 }) => {
+  const [selectedPersonaId, setSelectedPersonaId] = useState<string>("default");
+  const [customPersona, setCustomPersona] = useState<string>("");
+
+  // Memoize the calculation of the new instruction text based on UI state.
+  const newInstructionText = React.useMemo(() => {
+    const selected = PREDEFINED_PERSONAS.find(
+      (p) => p.id === selectedPersonaId,
+    );
+    if (!selected) return ""; // Should not happen, but a safe fallback.
+
+    return selected.id === "custom"
+      ? customPersona
+      : selected.systemInstruction;
+  }, [selectedPersonaId, customPersona]);
+
+  // Effect to sync UI changes (from dropdown or textarea) UP to the parent state.
+  useEffect(() => {
+    setGenerativeParams((prevParams) => {
+      const currentInstructionText =
+        prevParams.systemInstruction?.parts?.[0]?.text ?? "";
+
+      // Only update if the text content has actually changed to prevent re-renders.
+      if (newInstructionText === currentInstructionText) {
+        return prevParams;
+      }
+
+      const newSystemInstruction = newInstructionText
+        ? { parts: [{ text: newInstructionText }] }
+        : undefined;
+
+      return {
+        ...prevParams,
+        systemInstruction: newSystemInstruction,
+      };
+    });
+  }, [newInstructionText, setGenerativeParams]);
+
+  // Effect to sync parent state changes DOWN to the local UI state.
+  // This ensures the UI reflects the state if it's changed elsewhere.
+  useEffect(() => {
+    const instructionText =
+      generativeParams.systemInstruction?.parts?.[0]?.text ?? "";
+
+    const matchingPersona = PREDEFINED_PERSONAS.find(
+      (p) => p.id !== "custom" && p.systemInstruction === instructionText,
+    );
+
+    if (matchingPersona) {
+      // A predefined persona matches the current instruction.
+      setSelectedPersonaId(matchingPersona.id);
+      setCustomPersona("");
+    } else {
+      // No predefined persona matches. It's either custom or the default empty state.
+      if (instructionText) {
+        // It's a custom persona.
+        setSelectedPersonaId("custom");
+        setCustomPersona(instructionText);
+      } else {
+        // It's the default empty state.
+        setSelectedPersonaId("default");
+        setCustomPersona("");
+      }
+    }
```

Comment:
We should not set the persona back to default in the `else` branch here.

If a user selects 'Custom' persona, writes some `instructionText`, then deletes it to make it empty again, the selector reverts back to "Default" and the text box disappears. Instead the text box should stay open until the user explicitly selects a non-custom persona.

Reply:
Updated accordingly and successfully verified the custom field can be cleared and reentered.  thanks for your help!